src/test/java/bio/terra/service/snapshot/SnapshotConnectedTest.java

```diff
@@ -635,6 +635,98 @@ public void testExcludeLockedFromSnapshotFileLookups() throws Exception {
         connectedOperations.getDatasetExpectError(datasetRefSummary.getId(), HttpStatus.NOT_FOUND);
     }
 
+    @Test
+    public void testSnapshotArrayFileRefLookups() throws Exception {
+        // create a dataset
+        DatasetSummaryModel datasetRefSummary = createTestDataset("simple-with-array-filerefs-dataset.json");
+
+        // ingest a file
+        URI sourceUri = new URI("gs", "jade-testdata", "/fileloadprofiletest/1KBfile.txt",
+            null, null);
+        String targetFilePath =
+            "/mm/" + Names.randomizeName("testdir") + "/testExcludeLockedFromSnapshotFileLookups.txt";
+        FileLoadModel fileLoadModel = new FileLoadModel()
+            .sourcePath(sourceUri.toString())
+            .description("testExcludeLockedFromSnapshotFileLookups")
+            .mimeType("text/plain")
+            .targetPath(targetFilePath)
+            .profileId(billingProfile.getId());
+        FileModel fileModel = connectedOperations.ingestFileSuccess(datasetRefSummary.getId(), fileLoadModel);
+
+        // generate a JSON file with the fileref
+        String jsonLine = "{\"name\":\"name1\", \"file_ref\":[\"" + fileModel.getFileId() + "\"]}\n";
+
+        // load a JSON file that contains the table rows to load into the test bucket
```

Comment:
Could you make use of the `loadJsonData` function here? https://github.com/DataBiosphere/jade-data-repo/blob/cdcc92f6198fabb618ccecab80db2f72e0865133/src/test/java/bio/terra/service/snapshot/SnapshotConnectedTest.java#L753

Reply:
I could---I think I'd still have to override the file_ref column value to be the newly minted file id, so this doesn't seem that much messier